app/components/common/textBlurb/textBlurbDirective.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * textBlurbDirective.js
+ *
+ * A text box with a heading, and a read more link that expands the box to show the full contents
+ **/
+(function() {
+  'use strict';
+  
+  textBlurbDirective.$inject = ['$window'];
+  function textBlurbDirective($window) {
+    var directive = {
+      restrict: 'E',
+      bindToController: {
+        title: '@',
+        body: '@',
+      },
+      link: link,
+      controller: function() {},
+      controllerAs: 'textBlurb',
+      template: require('./textBlurb.html')
+    };
+
+    return directive;
+
+    //////////////////////////////
+    function link(scope, element, attrs, textBlurb) {
+      textBlurb.isCollapsed = true;
+      
+      scope.$watch(function() {return textBlurb.body; }, refreshButtons);
+      angular.element($window).bind('resize', refreshButtons);
+
+      function refreshButtons() {
+        textBlurb.showButtons = shouldShowButtons();
+        scope.$digest(); //since you're outside of angular, you need this here to update the view 
```

Comment:
I think this is just about where you've called the event to occur.  Pretty much all events are accessible within Angular, but the link function only executes when a component first compiles, and then isn't called again, which is why you have to manually call the digest cycle to handle the change.  If the window resize thing isn't working, try moving the function into a controller and see if that helps.